spark/v3.3/spark/src/main/java/org/apache/iceberg/spark/SparkCatalog.java

```diff
@@ -722,6 +741,9 @@ private Table loadFromPathIdentifier(PathIdentifier ident) {
       long snapshotIdAsOfTime = SnapshotUtil.snapshotIdAsOfTime(table, asOfTimestamp);
       return new SparkTable(table, snapshotIdAsOfTime, !cacheEnabled);
 
+    } else if (branch != null && table.snapshot(branch) != null) {
+      return new SparkTable(table, table.snapshot(branch).snapshotId(), !cacheEnabled);
```

Comment:
If the branch is non-null but the table doesn't have a branch with that name, then I think this must fail rather than returning the `else` branch, right?

Reply:
+1